driver/makedepend.ml

```diff
@@ -103,21 +103,15 @@ let find_module_in_load_path name =
     let uname = Unit_info.normalize name in
     List.map (fun ext -> uname ^ ext) (!mli_synonyms @ !ml_synonyms)
   in
-  let rec find_in_array a pos =
-    if pos >= Array.length a then None else begin
-      let s = a.(pos) in
-      if List.mem s names || List.mem s unames then
-        Some s
-      else
-        find_in_array a (pos + 1)
-    end in
   let rec find_in_path = function
-    [] -> raise Not_found
-  | (dir, contents) :: rem ->
-      match find_in_array contents 0 with
-        Some truename ->
-          if dir = "." then truename else Filename.concat dir truename
-      | None -> find_in_path rem in
+    | [] -> raise Not_found
+    | (dir, contents) :: rem ->
+        let mem s = List.mem s names || List.mem s unames in
+        match Array.find_opt mem contents with
```

Comment:
What about the following:

```ocaml
| (dir, contents) :: rem ->
  let tbl = Hashtbl.create 512 in
  List.iter (fun s -> Hashtbl.replace tbl s ()) names;
  List.iter (fun s -> Hashtbl.replace tbl s ()) unames;
  let mem = Hashtbl.mem tbl in
  match Array.find_opt mem contents with
  ...
```

?

Reply:
I don't think it matters much, the whole section is overridden by the commit [ocamldep: use Load_path](https://github.com/ocaml/ocaml/pull/12576/commits/327b58d3f5d2b52b57a36bd0291c458376605421) later on. I was told about the `Load_path` module after I wrote most of the code… so I just noticed I should have removed the [ocamldep: use Array.find_opt](https://github.com/ocaml/ocaml/pull/12576/commits/d4ba8837f1a950c65eaf7810e353fa5978475a56) commit from this PR.
If you have an objection against switching to `Load_path`, that's another issue.

Reply:
Yes indeed, I had a look commit/commit and didn't noticed at first. :)